<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="example">

  <!--CAMERA CRAFT-->
  <link name="kinect_camera">

    <visual>
      <origin rpy="0 0 0" xyz="-0.04 -0.03 0.08"/>
      <geometry>
        <mesh filename="package://gazebo_senai/meshes/kinect/meshes/kinect.dae"/>
      </geometry>
    </visual>

    <collision>
      <origin rpy="0 0 0" xyz="-0.04 -0.03 0.08"/>
      <geometry>
        <mesh filename="package://gazebo_senai/meshes/kinect/meshes/kinect.dae"/>
      </geometry>
    </collision>

    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.002"/>
      <inertia ixx="1.397e-7" ixy="0" ixz="-2.070e-12" iyy="1.512e-7" iyz="1.844e-9" izz="7.943e-8"/>
    </inertial>
  </link>


   
  <!--CAMERA SENSOR-->
  <gazebo reference="kinect_camera">
    <sensor type="camera" name="camera_top">
      <update_rate>30.0</update_rate>
      <camera name="head">
        <horizontal_fov>1.3962634</horizontal_fov>
        <image>
          <width>1366</width>
          <height>768</height>
          <format>R8G8B8</format>
        </image>
        <clip>
          <near>0.02</near>
          <far>300</far>
        </clip>
      </camera>
      <plugin name="camera_controller" filename="libgazebo_ros_camera.so">
        <alwaysOn>true</alwaysOn>
        <updateRate>0.0</updateRate>
        <cameraName>diff/camera_top</cameraName>
        <imageTopicName>image_raw</imageTopicName>
        <cameraInfoTopicName>camera_info</cameraInfoTopicName>
        <frameName>Camera</frameName>
        <hackBaseline>0.07</hackBaseline>
        <distortionK1>0.0</distortionK1>
        <distortionK2>0.0</distortionK2>
        <distortionK3>0.0</distortionK3>
        <distortionT1>0.0</distortionT1>
        <distortionT2>0.0</distortionT2>
      </plugin>
      <material>Gazebo/White</material>
    </sensor>
  </gazebo>


</robot>